data/json/vehicleparts/wheel.json

```diff
@@ -15,7 +15,14 @@
     "rolling_resistance": 0.375,
     "wheel_type": "rigid",
     "contact_area": 80,
-    "flags": [ "WHEEL" ]
+    "flags": [ "WHEEL", "RAIL" ]
+  },
+  {
+    "id": "metal_wheel_steerable",
+    "copy-from": "metal_wheel",
+    "type": "vehicle_part",
+    "name": "metal wheel (steerable)",
+    "extend": { "flags": [ "STEERABLE" ] }
   },
```

Comment:
I'd really prefer to see a new rail wheel, something like:
```
  {
    "id": "rail_wheel",
    "type": "vehicle_part",
    "name": "railroad wheel",
    "item": "wheel_rail",
    "location": "under",
    "symbol": "|",
    "broken_symbol": "x",
    "difficulty": 2,
    "durability": 360,
    "description": "A strong metal wheel.  A flange helps keep it on a rail, but makes it perform terribly when not on a rail.",
    "damage_modifier": 70,
    "breaks_into": [ { "item": "scrap", "count": [ 4, 6 ] }, { "item": "steel_chunk", "count": [ 2, 3 ] } ],
    "rolling_resistance": 0.25,
    "wheel_type": "rail",
    "contact_area": 80,
    "flags": [ "WHEEL", "RAIL" ]
  },
```
and add a rail wheel definition to src/veh_type.cpp:
```
static const std::vector<std::pair<std::string, int>> rail_terrain_mod = {{
        { "RAIL", 8 }
    }
};
```
and adjust vpart_info::load_wheel in the same file:
```
        } else if( wheel_type == "rail" ) {
            wh_info.terrain_mod = rail_terrain_mod;
            wh_info.or_rating = 0.05;
        }
```

Reply:
Oh well, thats how i did it initially, but decided to simplify it. Even the names matched. Will fix

Reply:
Also I've chosen metal wheel because I saw it on Passenger Car on train vehicle, should I really create new wheel? What about previous then?

Reply:
yes, you should create the new wheel.
and you should update the vehicle prototypes to use it.

Reply:
I did that, but old metal_wheel is used only in some spawnlists. 
Also I don't understand how to correctly add penalties for moving on `ROAD`for example, I increase `{ "RAIL", 8 }` value, but it looks like nothing changes, it can go full speed on road. Meanwhile on rails it can go only slow. How to correctly do that?